test/AppNotificationTests/AppNotificationHelpers.h

```diff
@@ -0,0 +1,237 @@
+            auto scopeExitGetAll = wil::scope_exit(
+                [&] {
+                    getAllAsync.Cancel();
+                });
+
+            VERIFY_ARE_EQUAL(getAllAsync.wait_for(std::chrono::milliseconds(500)), winrt::Windows::Foundation::AsyncStatus::Completed);
+            scopeExitGetAll.release();
+
+            auto actualToastVector{ getAllAsync.GetResults() };
+            if (actualToastVector.Size() == expectedToastVector.size())
+            {
+                for (int actualToastIndex{ 0 }; (uint32_t)actualToastIndex < actualToastVector.Size(); actualToastIndex++)
+                {
+                    for (int expectedToastIndex{ 0 }; expectedToastIndex < expectedToastVector.size(); expectedToastIndex++)
+                    {
+                        if (actualToastVector.GetAt(actualToastIndex).Id() == expectedToastVector[expectedToastIndex].Id())
+                        {
+                            VERIFY_ARE_EQUAL(actualToastVector.GetAt(actualToastIndex).Payload(), expectedToastVector[expectedToastIndex].Payload());
+                            break;
+                        }
+                    }
+                }
+
+                result = true;
+                break;
+            }
+
+            Sleep(500);
+        }
+        return result;
+    }
+
+    void PostToastHelper(std::wstring const& tag, std::wstring const& group)
+    {
+        winrt::AppNotification toast{ CreateToastNotification() };
+
+        toast.Tag(tag.c_str());
+        toast.Group(group.c_str());
+
+        winrt::AppNotificationManager::Default().Show(toast);
+
+        std::vector<winrt::AppNotification> toastVector{ toast };
+        VERIFY_IS_TRUE(VerifyToastsPosted(toastVector));
+    }
+
+    bool VerifyToastsCleared()
+    {
+        bool result{ false };
+        for (int i{ 0 }; i < 5; i++)
+        {
+            auto getAllAsync{ winrt::Microsoft::Windows::AppNotifications::AppNotificationManager::Default().GetAllAsync() };
+            auto scopeExitGetAll = wil::scope_exit(
+                [&] {
+                    getAllAsync.Cancel();
+                });
+
+            VERIFY_ARE_EQUAL(getAllAsync.wait_for(c_timeout), winrt::Windows::Foundation::AsyncStatus::Completed);
+            scopeExitGetAll.release();
+
+            auto actualToastVector{ getAllAsync.GetResults() };
+            if (actualToastVector.Size() == (uint32_t) 0)
+            {
+                result = true;
+                break;
+            }
+
+            Sleep(500);
+        }
+        return result;
+    }
+
+    bool EnsureNoActiveToasts()
+    {
+        auto removeAllAsync = winrt::AppNotificationManager::Default().RemoveAllAsync();
+        if (removeAllAsync.wait_for(c_timeout) != winrt::Windows::Foundation::AsyncStatus::Completed)
+        {
+            removeAllAsync.Cancel();
+            THROW_HR_MSG(E_FAIL, "Failed to remove all active toasts");
```

Comment:
`E_FAIL` is a bit blandly misleading. Should be `E_TIMEOUT` as that accurately conveys the error. Or _maybe_ `E_UNEXPECTED` in this case.

If this times out what's `wait_for()` return? If `AsyncStatus::Error` is that the only reason it could fail? If not, how do you know if it failed due to timeout vs other reasons?

P.S. `E_FAIL` is the worst error code, in addition to the obvious useless-tease ("Yeah, something went wrong, but not gonna tell you what, nyah, nyah") it's in various places across Windows, which can complicate troubleshooting if it ever occurs it can be vague what actually happened or why. Strive to never use/return/throw `E_FAIL`.